org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/lookup/InferenceContext18.java

```diff
@@ -2005,6 +2010,25 @@ public ReferenceBinding inferRecordPatternParameterization(
 		return getRecordPatternTypeFromUpwardsProjection(typeBinding, alphas, solution);
 	}
 
+	public void addConstraintsFromNestedPatterns(RecordPattern recordPattern) {
+		if (recordPattern.patterns != null) {
+			Pattern[] patterns = recordPattern.patterns;
+			RecordComponentBinding[] components = ((ReferenceBinding) recordPattern.resolvedType.original()).components();
```

Comment:
calling `original()` probably is cheating. I guess we need to peel raw types but not parameterized types.